### ISSUE TITLE
Strange RMG generation

### ISSUE THREAD

**Issue by nullkiller** [vcmi/vcmi#933]
![image](https://user-images.githubusercontent.com/5644734/190872353-26df5539-5b2d-4c0f-a6f4-e188598e423d.png)
![image](https://user-images.githubusercontent.com/5644734/190872378-a06e5408-e85a-4912-8410-aaa0543898b2.png)
[VCMI_Server_log.txt](https://github.com/vcmi/vcmi/files/9592429/VCMI_Server_log.txt)
[VCMI_Client_log.txt](https://github.com/vcmi/vcmi/files/9592430/VCMI_Client_log.txt)

**Comment by nullkiller** [comment]
hota + vcmi-extras and nothing else.

**Comment by misiokles** [comment]
This can be 'a feature'. Midnight Mix templates are 'strange' or strange conversion for vcmi format.

Original templates are here: http://heroes-iii.com/templ/midnightmix.txt

**Comment by nullkiller** [comment]
It looks like treasure placer got error. and did not put any obstacles. The rest of map is ok. Btw it is PR 902 codebase.

**Comment by Nordsoft91** [comment]
@misiokles is there documentation like this https://h3hota.com/en/templates?

This txt doesn't look human-readable:)

**Comment by DjWarmonger** [comment]
> This can be 'a feature'. Midnight Mix templates are 'strange' or strange conversion for vcmi format.
> 
> Original templates are here: http://heroes-iii.com/templ/midnightmix.txt

Nah, as far as I know they tend to generate nice, balanced maps.
Didn't test all there are one by one, though. Might look into #25 specifically.

**Comment by DjWarmonger** [comment]
In 1.3, RMG should add some extra obstacles if treasure density is very low.

**Comment by misiokles** [comment]
Can be closed as resolved.
